clientapi/routing/keys.go

```diff
@@ -32,9 +37,55 @@ func QueryKeys(
 	}
 }
 
-func UploadKeys(req *http.Request) util.JSONResponse {
+type uploadKeysRequest struct {
```

Comment:
Ultimately this is fine for now but I'd really love to see more of these types going into matrix-org/gomatrix eventually.

Reply:
Ideally we'd automate mapping swagger -> Go structs, but we're nowhere near that yet.